src/EnergyPlus/SwimmingPool.cc

```diff
@@ -1056,23 +1056,22 @@ void UpdatePoolSourceValAvg(EnergyPlusData &state, bool &SwimmingPoolOn) // .TRU
     // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
     SwimmingPoolOn = false;
 
-    // If this was never allocated, then there are no radiant systems in this input file (just RETURN)
+    // If this was never allocated, then there are no swimming pools in this input file (just RETURN)
     for (int PoolNum = 1; PoolNum <= state.dataSwimmingPools->NumSwimmingPools; ++PoolNum) {
         if (!allocated(state.dataSwimmingPools->Pool(PoolNum).QPoolSrcAvg)) return;
 
-        // If it was allocated, then we have to check to see if this was running at all
+        // If it was allocated, then we have to check to see if this pool was running at all.  If so, update pool terms also.
         for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
             if (state.dataSwimmingPools->Pool(PoolNum).QPoolSrcAvg(SurfNum) != 0.0) {
                 SwimmingPoolOn = true;
+                state.dataHeatBalFanSys->QPoolSurfNumerator(SurfNum) = state.dataSwimmingPools->Pool(PoolNum).QPoolSrcAvg(SurfNum);
+                state.dataHeatBalFanSys->PoolHeatTransCoefs(SurfNum) = state.dataSwimmingPools->Pool(PoolNum).HeatTransCoefsAvg(SurfNum);
                 break; // DO loop
             }
         }
-
-        state.dataHeatBalFanSys->QPoolSurfNumerator = state.dataSwimmingPools->Pool(PoolNum).QPoolSrcAvg;
-        state.dataHeatBalFanSys->PoolHeatTransCoefs = state.dataSwimmingPools->Pool(PoolNum).HeatTransCoefsAvg;
     }
```

Comment:
@RKStrand this code is different now. Before the entire array was copied and now only the first surface with a non-zero value is copied (because of the break). Is that intentional?

Reply:
@rraustad Thanks for taking a look at this.  This is all pretty subtle and because I'm also working on the ZeroSource and associated arrays, I was confused/concerned as soon as I saw your post.  In this case, because this is contained within a loop over all of the pools, this is actually correct for this bug.  There is only one surface that is valid for each pool (this should never been a loop in all honesty).  The problem with the array copy is that for any additional pools, the array copy overwrites valid information from the last pool which led to zeroes for vital, valid, non-zero information and causes the issue that the users noticed in this defect.  Basically, if you have more than one pool in an input file, then the last one always overwrites all the other ones and the pools do not control properly.

However, I am very glad you brought this up because with my work on the ZeroSource issue (#9926), I will need to be VERY careful about merging this file.  In fact, now that QPoolSrcAvg and HeatTransCoefsAvg should be a single value rather than an array, the looping over the surfaces is definitely not needed so that break should go away.  I will clean that up with the ZeroSource changes (no more loop).  Sound good?  I plan on flagging you for the potential review of #9926 so you can verify that I didn't forget.